lib/ecto/query/builder.ex

```diff
@@ -218,6 +219,12 @@ defmodule Ecto.Query.Builder do
     error! "`#{Macro.to_string(other)}` is not a valid query expression"
   end
 
+  def runtime_fragment!([{_,_}|_] = kw) when is_list(kw), do: kw
+  def runtime_fragment!(expr) do
+    raise ArgumentError, "to prevent sql injection, only a keyword list may be interpolated " <>
```

Comment:
That't what I thought initially – `ArgumentError` should be in runtime, but [there is what made me think other way about](https://github.com/elixir-lang/ecto/blob/da82346505b6182ba62b7d560df30599dbe81e4c/lib/ecto/query/builder/update.ex#L154-L160).
Then we should fix those calls instead (not related to the current PR). :smile: